ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -572,6 +572,8 @@ void markAsDeleted(StoreKey id, FileSpan fileSpan) throws StoreException {
     newValue.setNewOffset(fileSpan.getStartOffset());
     newValue.setNewSize(fileSpan.getEndOffset().getOffset() - fileSpan.getStartOffset().getOffset());
     addToIndex(new IndexEntry(id, newValue, null), fileSpan);
+    return new IndexValue(newValue.getSize(), newValue.getOffset(), newValue.getFlags(), newValue.getExpiresAtMs(),
```

Comment:
I think it's ok not to since in the put path, the index uses the value created in the store (i.e. by the caller). So there is already implicit trust.